datasets/sg/dev_board/crawler.py

```diff
@@ -0,0 +1,71 @@
+import re
+
+from zavod import Context, helpers as h
+from zavod.logic.pep import categorise
+from zavod.shed.zyte_api import fetch_html
+
+
+LINKS = {
+    # Mapping URLs to their respective position names
+    "https://www.edb.gov.sg/en/about-edb/our-team/executive-management.html": "Member of the Executive Committee",
+    "https://www.edb.gov.sg/en/about-edb/our-team/board-members.html": "Board Member",
+    "https://www.edb.gov.sg/en/about-edb/our-team/international-advisory-council.html": "Member of the International Advisory Council",
+}
+TITLE_REGEX = re.compile(r"^(Mr|Ms|Miss|Prof|Dr)\.? (?P<name>.+)$")
+
+
+def unblock_validator(doc) -> bool:
+    return doc.find(".//section[@class='container']") is not None
+
+
+def emit_person(context: Context, name, role, link, title, position_name):
+    person = context.make("Person")
+    person.id = context.make_id(name, role)
+    person.add("name", name)
+    person.add("title", title)
+    person.add("position", role)
+    person.add("topics", "role.pep")
+    if link is not None:
+        person.add("sourceUrl", link)
+
+    position = h.make_position(
+        context,
+        name=position_name,
+        country="sg",
+        topics=["gov.executive", "gov.national"],
+    )
+
+    categorisation = categorise(context, position, is_pep=True)
+    if categorisation.is_pep:
+        occupancy = h.make_occupancy(context, person, position)
+        if occupancy:
+            context.emit(person, target=True)
+            context.emit(position)
+            context.emit(occupancy)
+
+
+def crawl(context: Context):
+    for url, position_name in LINKS.items():
+        doc = fetch_html(context, url, unblock_validator, cache_days=3)
+        doc.make_links_absolute(url)
+        main_containers = doc.findall(".//section[@class='profile-listing analytics ']")
+
+        # Loop through each section to find and process the profiles
+        for section in main_containers:
+            profiles = section.findall(".//div[@class='profile-content-container']")
+            for profile in profiles:
+                # Extract the name, role, and link
+                name = profile.find(".//h5").text_content().strip()
+                name_match = TITLE_REGEX.match(name)
+                if name_match:
+                    name = name_match.group("name")
+                    title = name_match.group(1)
+                else:
+                    context.log.warn(f"Could not extract title from name: {name}")
```

Comment:
did you split the "emit" part to its own function because this was getting a bit long? Or some other reason?

we normally prefer a "crawl_person" which extracts and emits the data in one place.

In this case I think there's a bug where if one iteration of the loop fails to extract title, the previous title will be used in that iteration. That sort of bug is less likely when each person is handled in their own function call because the title variable won't be initialised if it doesn't match.

Reply:
https://zavod.opensanctions.org/best_practices/patterns/#common-crawler-code-structure